src/setuptools_scm/integration.py

```diff
@@ -1,14 +1,48 @@
+import os
 import warnings
 
 import setuptools
 
 from . import _get_version
-from . import Configuration
+from .config import _read_dist_name_from_setup_cfg
+from .config import Configuration
 from .utils import do
 from .utils import iter_entry_points
 from .utils import trace
 
 
+def _warn_on_old_setuptools(_version=setuptools.__version__):
+    if int(_version.split(".")[0]) < 45:
```

Comment:
Is this 31 or 42? Does this only get triggered on the pyproject.toml codepath? Should not be 45, at least, that's only required to build setuptool_scm itself, IIUC.

Reply:
setuptools 45 is the target intended minimal version (i dont plan to loosen that up)